src/taskgraph/main.py

```diff
@@ -556,6 +558,22 @@ def load_image(args):
         sys.exit(1)
 
 
+def validate_docker():
+    try:
+        p = subprocess.run(["docker", "ps"])
+        if p.returncode != 0:
+            print(
+                "Error: Cannot connect to the Docker daemon. Is the docker daemon running?"
+            )
+            sys.exit(1)
+    except Exception:
+        traceback.print_exc()
+        print(
+            "Error: Cannot connect to the Docker daemon. Is the docker daemon running?"
+        )
+        sys.exit(1)
+
```

Comment:
There shouldn't be any need to catch `Exception` here. By default the `subprocess.run` function has `check=False`, which means it won't raise an exception if the command fails. If we *do* somehow get an exception here, it's almost certainly not due to docker being stopped, so we should let it propagate.